plugin-server/src/worker/ingestion/timestamps.ts

```diff
@@ -4,24 +4,32 @@ import { DateTime, Duration } from 'luxon'
 
 import { status } from '../../utils/status'
 
-type InvalidTimestampCallback = () => void
+type InvalidTimestampCallback = (field: string, value: string, reason: string) => void
 
 export function parseEventTimestamp(data: PluginEvent, callback?: InvalidTimestampCallback): DateTime {
```

Comment:
the code for these functions looks a little complex now I wonder if it can be simplified